extensions/ima/src/main/java/com/google/android/exoplayer2/ext/ima/AdTagLoader.java

```diff
@@ -1122,12 +1122,17 @@ private void handleAdPrepareError(int adGroupIndex, int adIndexInAdGroup, Except
   }
 
   private void ensureSentContentCompleteIfAtEndOfStream() {
-    if (!sentContentComplete
-        && contentDurationMs != C.TIME_UNSET
-        && pendingContentPositionMs == C.TIME_UNSET
-        && getContentPeriodPositionMs(checkNotNull(player), timeline, period)
-                + THRESHOLD_END_OF_CONTENT_MS
-            >= contentDurationMs) {
+
+    if (sentContentComplete
+        || contentDurationMs == C.TIME_UNSET
+        || pendingContentPositionMs != C.TIME_UNSET) {
+      return;
+    }
+
+    long contentPeriodPositionMs = getContentPeriodPositionMs(checkNotNull(player), timeline, period);
+    int pendingAdGroupIndex = adPlaybackState.getAdGroupIndexAfterPositionUs(contentPeriodPositionMs, contentDurationMs);
```

Comment:
Should this be getAdGroupIndexForPositionUs? I assume there is no ad group following in the case where the user seeks to the end of the content. 

Otherwise, looks great! Thanks for making all these additional fixes :)

Reply:
I thought so too, but `getAdGroupIndexForPositionUs` returns a -1 for some reason.

I did test Midroll and Midroll + Postroll using this code and it works as expected.

Reply:
That's because the position you pass into the method is in milliseconds but the method expects microseconds. I'll make this update when merging the change internally. Thanks!

Reply:
Oooh, missed that. Thanks for pointing it out. Just tested it, that fixes the issue.
I can update the PR again using the `Util.msToUs` method, should I? Or would you suggest any other change to the `getAdGroupIndexForPositionUs` method itself?

Reply:
No worries about that. I already imported the PR to our internal code and will make the update there. You should see the merge here (including my additional changes) once we sync back to GitHub.

Reply:
Great. Happy to contribute :)